PS-5695: Reformat scrubbing code (5.7)

dutow opened the pull request.

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -49,54 +50,54 @@ uint srv_background_scrub_data_interval = (7 * 24 * 60 * 60);
-	btr_scrub_t* scrub_data, /*!< in: data to store statistics on */
-	buf_block_t* block,	 /*!< in: block */
-	dberr_t err)             /*!< in: error */
+    /*===============*/
```
The old-style function header comments format poorly. Not sure if removing them is reasonable to ask in this PR (this would also entail moving /*!< in ... */ arg comments to function header comment

dutow replied:
I've updated the comments, will do the same for 8.0.

laurynas-biveinis replied:
"/*====*/" should go away too
```suggestion
```

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) 2017, MariaDB Corporation.
 
-/**************************************************//**
+/**************************************************/ /**
```
Optional but if you have to repush due to other comments
```suggestion
/**
```

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -49,54 +50,57 @@ uint srv_background_scrub_data_interval = (7 * 24 * 60 * 60);
-void
+/** Log a scrubbing failure
+@param[in]	index
+@param[in,out]	scrub_data
```
Lost comment on move. There are some more, I am only pointing out those which add something more than arg name (no point in having "param[in] block block" comments)
```suggestion
@param[in,out]	scrub_data data to store statistics on
```

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -277,88 +265,74 @@ btr_page_needs_scrubbing(
-*/
-
+/** Handle a skipped page
+@param[in]	scrub_data
```
Lost comments on move
```suggestion
@param[in]	scrub_data  data with scrub state
```

dutow replied:
I left these "data with scrub state" out intentionally - it doesn't explain more than scrub_data.

laurynas-biveinis replied:
OK

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -277,88 +265,74 @@ btr_page_needs_scrubbing(
-
+/** Handle a skipped page
+@param[in]	scrub_data
+@param[in]	needs_scrubbing */
```
Lost comments on move
```suggestion
@param[in]	needs_scrubbing   return code from btr_page_needs_scrubbing */
```

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -216,17 +205,16 @@ check_scrub_setting(
+@param[in]	allocated	is block known to be allocated
+@return         BTR_SCRUB_PAGE if page should be scrubbed
+                else BTR_SCRUB_SKIP_PAGE should be called
+                 with this return value (and without any latches held) */
```
Formatting

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -781,44 +719,34 @@ btr_scrub_page(
-Start iterating a space */
-
+/** Start iterating a space
+@param[out]	scrub_data
```
param[in] space

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -855,42 +780,32 @@ btr_scrub_update_total_stat(btr_scrub_t *scrub_data)
-/*********************************************************************
-Return scrub statistics */
+/* Return scrub statistics
+@param[out]	stat	 scrub data */
```
```suggestion
@param[out]	stat	 scrub statistics */
```

laurynas-biveinis commented on storage/innobase/include/btr0scrub.h:
```diff
@@ -58,98 +57,82 @@ struct btr_scrub_t {
+/** Check if a page needs scrubbing
+@param[in] 	scrub_data
+@param[in]	block
+@param[in]	allocated	is block allocated
```
```suggestion
@param[in]	allocated	is block allocated, free, or unknown
```

laurynas-biveinis commented on storage/innobase/include/btr0scrub.h:
```diff
@@ -58,98 +57,82 @@ struct btr_scrub_t {
-*/
+@param[in] 	scrub_data
+@param[in]	block
+@param[in]	allocated	is block allocated
```
```suggestion
@param[in]	allocated	is block allocated or free
```

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -216,17 +202,16 @@ check_scrub_setting(
+/** Check if a page needs scrubbing 
+@param[in]	scrub_data
+@param[in]	block		block to check, latched
+@param[in]	allocated	is block allocated, free, or unknown
```
Comment changed from original, hopefully OK

dutow replied:
it is, unknown was mentioned in the header but not in the cc file

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -277,88 +262,74 @@ btr_page_needs_scrubbing(
-
+/** Handle a skipped page
+@param[in]	scrub_data
+@param[in]	needs_scrubbing	return code from btr_page_needs_scrubbing*/
```
```suggestion
@param[in]	needs_scrubbing	return code from btr_page_needs_scrubbing */
```

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -499,18 +463,18 @@ btr_pessimistic_scrub(
-/*========*/
-	dict_table_t* table, /*!< in: table */
-	index_id_t index_id) /*!< in: index id */
+    /*========*/
```
```suggestion
```

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -499,18 +463,18 @@ btr_pessimistic_scrub(
-	dict_table_t* table, /*!< in: table */
-	index_id_t index_id) /*!< in: index id */
+    /*========*/
+    dict_table_t *table, /*!< in: table */
```
```suggestion
    dict_table_t *table,
```

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -499,18 +463,18 @@ btr_pessimistic_scrub(
-	index_id_t index_id) /*!< in: index id */
+    /*========*/
+    dict_table_t *table, /*!< in: table */
+    index_id_t    index_id) /*!< in: index id */
```
```suggestion
    index_id_t    index_id)
```

laurynas-biveinis commented on storage/innobase/include/btr0scrub.h:
```diff
@@ -58,98 +57,82 @@ struct btr_scrub_t {
-
-/****************************************************************
-Perform cleanup needed for a page not needing scrubbing */
+    /*============*/
```
```suggestion
```

laurynas-biveinis commented on storage/innobase/log/log0log.cc:
```diff
@@ -3013,90 +3013,83 @@ log_mem_free(void)
 ulonglong innodb_scrub_log_speed;
 
-/*****************************************************************//**
+/*****************************************************************/ /**
```
```suggestion
/**
```

dutow commented:
Updated. Also fixed pointer alignment and return statements manually as with redo encryption.

https://ps57.cd.percona.com/job/percona-server-5.7-param/97/

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -49,54 +49,55 @@ uint srv_background_scrub_data_interval = (7 * 24 * 60 * 60);
-static ib_mutex_t scrub_stat_mutex;
-mysql_pfs_key_t scrub_stat_mutex_key;
+static ib_mutex_t       scrub_stat_mutex;
+mysql_pfs_key_t		scrub_stat_mutex_key;
```
Should be aligned but don't repush if that's the sole comment

dutow replied:
And it looked aligned locally, but the last line used tabs, the other spaces. fixed.

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -445,22 +412,22 @@ btr_pessimistic_scrub(
-	ulint flags = BTR_MODIFY_TREE;
+	mem_heap_t *heap = NULL;
+	dtuple_t *  entry = NULL;
+	ulint *     offsets = NULL;
```
Probably *entry and *offsets

laurynas-biveinis commented on storage/innobase/btr/btr0scrub.cc:
```diff
@@ -445,22 +412,22 @@ btr_pessimistic_scrub(
 	* position a cursor on first record on page
 	*/
-	rec_t* rec = page_rec_get_next(page_get_infimum_rec(page));
+	rec_t *   rec = page_rec_get_next(page_get_infimum_rec(page));
```
*rec
